MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,455 @@
+                          .Select((e, i) => (Original: e.Identifier.ValueText, Alias: abbreviatedTypeNodes[i]))
+                          .ToDictionary(e => e.Original, e => e.Alias)
+                    select new
+                    {
+                        Syntax = md,
+                        Name = md.Identifier.ToString(),
+                        TypeParameterCount = md.TypeParameterList?.Parameters.Count ?? 0,
+                        TypeParameterAbbreviationByName = typeParameterAbbreviationByName,
+                        ParameterCount = md.ParameterList.Parameters.Count,
+                        SortableParameterTypes =
+                            from p in md.ParameterList.Parameters
+                            select CreateTypeKey(p.Type,
+                                                 n => typeParameterAbbreviationByName != null
+                                                   && typeParameterAbbreviationByName.TryGetValue(n, out var a) ? a : null),
+                    }
+                select m;
+
+        var q =
+            from e in qq.Select((m, i) => (SourceOrder: i + 1, Method: m))
+            orderby
+                e.Method.Name,
+                e.Method.TypeParameterCount,
+                e.Method.ParameterCount,
+                new TupleTypeKey(ImmutableList.CreateRange(e.Method.SortableParameterTypes))
+            select new
+            {
+                e.Method,
+                e.SourceOrder,
+            };
+
+            q = q.ToArray();
+
+            if (debug)
+            {
+                foreach (var m in
+                    from e in q
+                    let m = e.Method
+                    select new
+                    {
+                        m.Name,
+
+                        SourceOrder = e.SourceOrder.ToString("000", CultureInfo.InvariantCulture),
+
+                        TypeParameters =
+                            m.TypeParameterCount == 0
+                            ? string.Empty
+                            : "<" + string.Join(", ", from a in m.TypeParameterAbbreviationByName
+                                                      select a.Value) + ">",
+                        Parameters =
+                            "(" + string.Join(", ", m.SortableParameterTypes) + ")",
+
+                        Abbreviations =
+                            m.TypeParameterCount == 0
+                            ? string.Empty
+                            : " where " + string.Join(", ", from a in m.TypeParameterAbbreviationByName
+                                                            select a.Value + " = " + a.Key),
+                    }
+                    into e
+                    select e.SourceOrder + ": "
+                         + e.Name + e.TypeParameters + e.Parameters + e.Abbreviations)
+                {
+                    Console.Error.WriteLine(m);
+                }
+            }
+
+            var indent = new string('\x20', 4);
```

Comment:
Why not simply a `' '` ?

Reply:
Because I can't tell if `' '` is one or two spaces. 😛 But okay, this is not JavaScript and we know that apostrophe-quoting is for a single character so I can change that.